### PR TITLE
Add ADD_NAMESPACES option to support strict XML documents

### PR DESCRIPTION
**Pull request by kevin-deyoungster**
> This pull request implements feature #728: Support additional namespaces in XML documents 

### Background & Context 
DOMPurify cannot sanitize XML documents with custom namespaces. This PR introduces a CONFIG option `ADD_NAMESPACES` to enable users to allowlist custom namespaces to consider during sanitization.

#### Sample Usage
```js
DOMPurify.sanitize(dirty, {
    PARSER_MEDIA_TYPE: "application/xhtml+xml",
    ADD_TAGS: [],
    ADD_ATTR: [],
    ADD_NAMESPACES: []
})
```

### Tasks 
- [x] Add `ADD_NAMESPACES` config parameter
- [x] Confirm from reviewers if feature can be supported 
- [x] Refactor
- [x] Add to test suite: "Config-Flag tests: ADD_NAMESPACES"

### CODE REVIEW

**Comment by cure53** [comment]
Hey there, thanks for the PR, this looks generally good and I think supporting the feature as proposed makes sense!

**Comment by kevin-deyoungster** [comment]
> @cure53, I've modified the proposed changes after feedback from @tosmolka. 

### Update
Change `ADD_NAMESPACES`  config option ➜ `ALLOWED_NAMESPACES`


### Detail
If user provides the `ALLOWED_NAMESPACES` config option, they have to specify ALL valid namespaces. If it's not provided, it'll default to the current supported NAMESPACES (`HTML, SVG, MATHML`).


### Sample Usage
This does not change how proposed API changes will be used, only the CONFIG name

```js
DOMPurify.sanitize(dirty, {
    PARSER_MEDIA_TYPE: "application/xhtml+xml",
    ADD_TAGS: [],
    ADD_ATTR: [],
    ALLOWED_NAMESPACES: []
})
```

File: src/purify.js [hunk 1]
```diff
@@ -634,14 +647,18 @@ function createDOMPurify(window = getGlobal()) {
     // can be null. We just simulate parent in this case.
     if (!parent || !parent.tagName) {
       parent = {
-        namespaceURI: HTML_NAMESPACE,
+        namespaceURI: NAMESPACE,
```
Comment (cosmincatrinescu) [inline src/purify.js:650]: Hey, curious why refactor here but still use HTML_NAMESPACE on #343?

**Comment by kevin-deyoungster** [comment]
@cure53, could you kindly review, thanks!

**Review by cure53 — APPROVED** [review]

**Comment by kevin-deyoungster** [comment]
@cure53, awesome. 

Could you add me as contrib. so I can merge the PR in?
I don't have write permissions to this repo to trigger the merge.

**Comment by cure53** [comment]
@kevin-deyoungster Awesome, thanks :)

I just did the merge and yes, it's likely necessary to send a small PR to the DT folks, thanks a lot!

**Pull request merged**